src/frontend/js/tinymce-plugins/custom_link_input/plugin.js

```diff
@@ -0,0 +1,291 @@
+                    api.enable("submit");
+                } else {
+                    api.disable("submit");
+                }
+
+                // make ajax new ajax request on user input
+                if (data.search != prev_search_text && data.search != "") {
+                    ajax_request_id += 1;
+                    getCompletions(data.search, ajax_request_id).then(([new_completions, request_id]) => {
+                        if (request_id != ajax_request_id) return;
+
+                        completion_items.length = 0;
+                        for (const completion of new_completions) {
+                            completion_items.push({
+                                text: completion.title,
+                                value: completion.url,
+                            });
+                        }
+
+                        // It seems like there is no better way to update the completion list 
+                        api.redial(dialog_config);
+                        api.setData(data);
+                        api.focus("search");
+                        prev_search_text = data.search;
+                        updateDialog(api);
+                    });
+                } else if (data.search == "" && prev_search_text != "") {
+                    // force an update so that the original user url can get restored
+                    completion_items.length = 0;
+                    api.redial(dialog_config);
+                    api.setData(data);
+                    api.focus("search");
+                    prev_search_text = data.search;
+                    updateDialog(api);
+                }
+            }
+
+            const dialog_config = {
+                title: tinymceConfig.getAttribute("data-link-dialog-title-text"),
+                body: {
+                    type: "panel",
+                    items: [
+                        {
+                            type: "input",
+                            name: "url",
+                            label: tinymceConfig.getAttribute("data-link-dialog-url-text")
+                        },
+                        {
+                            type: "input",
+                            name: "text",
+                            label: tinymceConfig.getAttribute("data-link-dialog-text-text")
+                        },
+                        {
+                            type: "label",
+                            label: tinymceConfig.getAttribute("data-link-dialog-internal_link-text"),
+                            items: [
+                                {
+                                    type: "input",
+                                    name: "search",
+                                },
+                                {
+                                    type: "selectbox",
+                                    name: "completions",
+                                    items: completion_items
+                                }
```

Comment:
Do you think you could add a default (but disabled) option for the select field like " - no results - " or similar when there's nothing to select? Only if it's not too complicated :sweat_smile: